webhooks/virtualmachine/mutation/virtualmachine_mutator_unit_test.go

```diff
@@ -1099,4 +1100,44 @@ func unitTestsMutating() {
 			})
 		})
 	})
+
+	Describe("SetLastResizeAnnotation", func() {
+		const newClassName = "my-new-class"
+
+		var (
+			oldVM *vmopv1.VirtualMachine
+		)
+
+		BeforeEach(func() {
+			pkgcfg.UpdateContext(ctx, func(config *pkgcfg.Config) {
+				config.Features.VMResize = true
+			})
+			oldVM = ctx.vm.DeepCopy()
+		})
+
+		When("vm ClassName does not change", func() {
+			It("does not set last-resize annotation", func() {
+				updated, err := mutation.SetLastResizeAnnotation(&ctx.WebhookRequestContext, ctx.vm, oldVM)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updated).To(BeFalse())
+
+				_, _, _, exists := vmopv1util.GetLastResizedAnnotation(*ctx.vm)
+				Expect(exists).To(BeFalse())
+			})
+		})
+
+		When("existing vm ClassName changes", func() {
+			It("set last-resize annotation", func() {
+				ctx.vm.Spec.ClassName = newClassName
+
+				updated, err := mutation.SetLastResizeAnnotation(&ctx.WebhookRequestContext, ctx.vm, oldVM)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(updated).To(BeTrue())
+
+				className, _, _, exists := vmopv1util.GetLastResizedAnnotation(*ctx.vm)
+				Expect(exists).To(BeTrue())
+				Expect(className).To(Equal(oldVM.Spec.ClassName))
+			})
+		})
+	})
```

Comment:
Maybe worth adding another test case where the class name is changed, but the VM already has the last-resize annotation set (to validate that it doesn't overwrite the existing last-resize annotation value)?